ProcessOptimizer/tests/test_optimizer.py

```diff
@@ -328,3 +332,60 @@ def test_iterating_ask_tell_lhs():
     assert opt.ask(n_points=3) == samples[2:5]  # samples 2,3 and 4
     opt.tell([[2], [2], [2]], [0, 0, 0])
     assert opt.ask() == samples[5]  # sample 5
+
+
+@pytest.mark.slow_test
+def test_add_remove_modelled_noise():
+    '''
+    Tests whether the addition of white noise leads to predictions closer to
+    known true values of experimental noise (iid gaussian noise)'''
+    # Define objective function
+    def flat_score(x):
+        return 42
+    
+    # Set noise and model system
+    noise_size = 0.45
+    flat_space = [(-1.0, 1.0)]
+    flat_noise = {"model_type": "constant",
+                  "noise_size": noise_size,
+                  }
+    # Build ModelSystem object
+    model = ModelSystem(score=flat_score,
+                        space=flat_space,
+                        noise_model=flat_noise,
+                        )
+    # Instantiate Optimizer
+    opt = Optimizer(flat_space,
+                    "GP",
+                    lhs=False,
+                    n_initial_points=1,
+                    random_state=42
+                    )
+    #Make 20 dispersed points on X
+    next_x = np.linspace(-1, 1, 20).tolist()
+    x = []
+    y = []
+    #sample noisy experiments, 10 in each x-value
+    for i in range(10):
+        for xx in next_x:
+            x.append([xx])
+            y.append(model.get_score([xx]))
+    # Fit the model
+    res = opt.tell(x,y)
+    res_x, [res_y, res_std_no_white] = expected_minimum(res, return_std=True)
+    #Add moddeled experimental noise
+    opt.add_modelled_noise()
+    res_noise = opt.get_result()
```

Comment:
Would it make sense to do a copy of the entire optimizer in this case? 
In details: train the optimizer on the data. Make a copy of the optimizer; call it opt_noisy. Add the noise to apt_noisy. Then pull the std from the expected minima?
And now that I write this... shouldn't we check that the minima are identical in x and y (?)

Reply:
Well, the res_x location in your test is probably quite sensitive to the specific simulated data points (as there isn't an optimum) so I wouldn't want to add it for that reason alone.

More importantly, with the way our model is defined, the WhiteNoise is simply a constant offset to the uncertainty with the same value across the whole parameter space. Because of this, by definition it cannot change the location of the expected minimum in mean values. If the WhiteNoise is very large it may change the location in practice as a consequence of using sampling to find the expected minimum, but it doesn't add any information and so should be left out.